qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -1,22 +1,177 @@
 import re
 import textwrap
-from typing import Optional, TYPE_CHECKING, Tuple, Union
+import numpy as np
+from typing import Optional, TYPE_CHECKING, Tuple, Union, Any, Dict, cast, List
 
+from qcodes.instrument.parameter import MultiParameter
+from qcodes.instrument.group_parameter import GroupParameter, Group
 from qcodes.instrument.channel import InstrumentChannel
+import qcodes.utils.validators as vals
 
 from .KeysightB1500_module import B1500Module, parse_dcorr_query_response, \
     format_dcorr_response, _DCORRResponse
 from .message_builder import MessageBuilder
 from . import constants
-from .constants import ModuleKind, ChNr
+from .constants import ModuleKind, ChNr, MM
+
 if TYPE_CHECKING:
     from .KeysightB1500_base import KeysightB1500
 
+_pattern = re.compile(r"((?P<status>\w)(?P<chnr>\w)(?P<dtype>\w))?"
+                      r"(?P<value>[+-]\d{1,3}\.\d{3,6}E[+-]\d{2})")
+
+
+class CVSweep(InstrumentChannel):
```

Comment:
Maybe CVSweeper ? I would expect CVSweep to be a parameter containing the sweep data

Reply:
Sweeper sounds better. :)